### ISSUE TITLE
Non-bayesian estimation should use quasi-Maximum likelihood standard errors

### ISSUE THREAD

**Issue by tholden** [DynareTeam/dynare#1170]
At present, with non-Bayesian estimation, Dynare computes standard errors using the Hessian of the likelihood. This is only valid if it is assumed that the shocks in the "true" model are normally distributed. And, in that case, it is an inefficient way of computing the standard errors, as it will be equal to the Fisher information matrix, which only requires the calculation of the derivative of the score vector.

It would make more sense to default to computing quasi-Maximum likelihood "sandwich" covariances, with the option to use the Fisher information matrix if the user wants quicker results.

**Comment by MichelJuillard** [comment]
Do you have recent references on computing quasi-Maximum likelihood "sandwich" covariances?

**Comment by tholden** [comment]
Canova's textbook covers it (though it's not particularly clear), in the 2007 edition, it's the bottom of page 210, after exercise 6.17.
This textbook chapter covers it more clearly: http://homepage.ntu.edu.tw/~ckuan/pdf/et01/ch9.pdf
Plus these MIT notes cover it at the end: https://dspace.mit.edu/bitstream/handle/1721.1/46343/14-384Fall-2007/NR/rdonlyres/Economics/14-384Fall-2007/D6349EED-D2FE-441A-8BC4-C60043D83675/0/lec21.pdf

And you can see my own implementation of the calculation within DynareOBC at this line: https://github.com/tholden/dynareOBC/blob/master/dynareOBC/dynareOBCCore.m#L425

**Comment by stepan-a** [comment]
We had this in dynare 3... I do not remember why we removed it. I agree that this should be an option. I wonder if this estimator is not already available with ```mode_compute=5``` (@rattoma 's likelihood optimizer).

**Comment by rattoma** [comment]
yes, I think that I1 in  MIT notes page 5 is the default output "hessian" [outer product gradient] from `mode_compute=5`. So, this should be combined with I2 in the same notes [i.e. the hessian] to get the sandwich if I am right.

**Comment by JohannesPfeifer** [comment]
@MichelJuillard The reference is Theorem 3.2 of Halbert White (1982): Maximum Likelihood Estimation of Misspecified Models, Econometrica, Vol. 50, No. 1, pp. 1-25

**Comment by MichelJuillard** [comment]
Thanks

**Comment by JohannesPfeifer** [comment]
@houtanb Could you please assign this to @rattoma as discussed at the Dynare conference 2016

**Comment by paologelain** [comment]
It seems from tha manual that the default option for optimizer 5 is Hessian routine, which is not the one to get I1. Option "0" gives the outer product gradient.

**Comment by paologelain** [comment]
I confirm that the one can get the sandwich by using mode_compute = 5 (together with kalman_algo = 2 to trigger the univariate filter). One needs to use a recent dynare version. For instance version 4.4.3 seems to ignore the options of mode_compute = 5.
The formula that I know to get the asymptotic consistent and robust variance covariance matrix is
inv(A)*B*inv(A)
where A corresponds to I2 in the MIT note and B to I1
One can obtain A and B as follows
A = -hh/n; with hh the hessian estimated with optim=('Hessian',1) and n is the sample size
B = hh/n;  with hh the hessian estimated with optim=('Hessian',0) and n is the sample size
